webpack/assets/javascripts/react_app/components/SettingsTable/index.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { selectSettingsByCategory } from '../SettingRecords/SettingRecordsSelectors';
+
+import { setSettingToUpdate } from './SettingsTableActions';
+
+import SettingsTable from './SettingsTable';
+
+import reducer from './SettingsTableReducer';
+
+import useSettingModal from '../SettingUpdateModal/useSettingModal';
+
+export const reducers = {
+  settingsTable: reducer,
+};
+
+const WrappedSettingsTable = props => {
+  const settings = useSelector(state =>
+    selectSettingsByCategory(state, props.category)
+  );
+
+  const dispatch = useDispatch();
+  const { setModalOpen } = useSettingModal();
+
+  const onEditClick = async setting => {
+    await dispatch(setSettingToUpdate(setting));
+    setModalOpen();
```

Comment:
we could just set the `editing: <id>` by some action, and the modal would pick up on that, so we won't need to call `setModalOpen()` explicitly.

Reply:
We use hooks to open or close modals, they have their own slice of redux store:

![redux-modal](https://user-images.githubusercontent.com/2664090/88369828-23a21c00-cd91-11ea-86e0-12c48c3f3b5c.png)

ForemanModal does not currently support the behaviour that you describe.